src/formatDate.js

```diff
@@ -50,7 +50,37 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateList = date.split(fromFormat[3]);
+
+  // link old order with date
+  const dateObj = {};
+
+  dateList.forEach((elem, ind) => {
+    dateObj[fromFormat[ind]] = elem;
+  });
+
+  // format date
+  const formatList = [];
+
+  toFormat.forEach((elem, ind) => {
+    if (elem.length === 1) {
+      return false;
+    }
+
+    if (elem.length === 2 && !(elem in dateObj)) {
+      dateObj[elem] = dateObj[elem + elem].slice(2, 4);
+    } else if (elem.length === 4 && !(elem in dateObj)) {
```

Comment:
I think it`s can simplify your code
```suggestion
    if (elem === "YY" && !(elem in dateObj)) {
      dateObj[elem] = dateObj[elem + elem].slice(2, 4);
    } else if (elem === "YYYY" && !(elem in dateObj)) {
```